Peaky/ParameterSet.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Peaky
+{
+    internal class ParameterSet : IEnumerable<Parameter>
+    {
+        private readonly ImmutableDictionary<string, object> parameters;
+        private readonly string queryString;
+
+        public ParameterSet(IEnumerable<Parameter> parameters)
+        {
+            this.parameters = parameters.ToImmutableDictionary(e => e.Name, e => e.DefaultValue);
```

Comment:
Why not store as `Parameter`, rather than doing the transformation only in `GetEnumerator`?

Reply:
i do this to avoid duplicates for same key

Reply:
but .... you have a point! a method cannot have two parameters with the same name ....